.github/workflows/generate_pages.yml

```diff
@@ -0,0 +1,36 @@
+name: Build and push documentation for main branch
```

Comment:
This file is almost identical to ckeck_documentation_build.yaml.
You could have one yaml file, and then run step "Cleanup delete target branch" only if it's not the main branch. (See [here](https://github.com/exasol/script-languages-release-c4/blob/a7ff92274894c313691caeae5ecddc2a7451ef2d/.github/workflows/release.yml#L42) for an example)